src/Robo/Common/UserConfig.php

```diff
@@ -38,15 +38,9 @@ public function __construct(string $configDir) {
       $this->config = json_decode(file_get_contents($this->configPath), TRUE);
     }
     else {
-      if (!file_exists($configDir)) {
-        if (is_writable(dirname($configDir))) {
-          mkdir($configDir, 0777, TRUE);
-        }
-        else {
-          return;
-        }
+      if (!file_exists($configDir) && mkdir($configDir, 0777, TRUE)) {
```

Comment:
I think you're on the right path, but won't this fail to create the json file if for some reason the directory exists but the json file doesn't?

Reply:
Nice catch. Updated PR.